docs/libvirt-pod-networking.md

```diff
@@ -0,0 +1,117 @@
+# Using CNI Interfaces and libvirt for networking
+
+Author: Fabian Deutsch \<fabiand@redhat.com\>
+
+## Introduction
+
+Today networking in KubeVirt is not integrated with Kubernetes, a user can
+rather choose to create a specific VM Spec definition to create a particular
+libvirt domxml, to reuse an existing physical device.
+But this is obviously suboptimal
+
+- It is not integrated with Kubernetes
+- It does not provide connectivity with pods
+- It requires to specify host specific bits (the network device to use)
+
+The purpose of this document is to describe one approach of how we can provide
+a minimal integrated solution, with few workarounds.
+
+**Note:** This approach is rough, and not optimized, but the rule of thumb was
+to find a robust solution which works with Kubernetes today.
+
+
+## Use-case
+
+The primary use-case is to allow VM to get connected to the pod network.
+
+
+## API
+
+The API is pretty clear: As usual an interface will be defined.
+To indicate that Pod interfaces shall be used for networking, a new type
+`pod-network` is introduced.
+All other fields, concerning the properties of the virtual NIC, can be used
+as usual.
+
+```yaml
+kind: VM
+spec:
+  domain:
+    devices:
+      interfaces:
+      - type: pod-network
+        mac:
+          address: 00:11:22:33:44:55
+        model:
+          type: virtio
+```
+
+
+## Implementation
+
+Prerequisites
+- libvirt is using Pod networking
+- virt-handler and libvirt are running in the same Pod or rather network
```

Comment:
I would prefer namespace switches in virt-handler, or in code along-side virt-handler (like cni plugins do), over having a separate daemon just because of network namespaces. CNI plugins change namespaces, the kubelet changes and creates namespaces, it is not black magic in my eyes, which we need to avoid no matter what it costs. But right now, virt-handler can just stay in the network namespace.

I would rather want to avoid having X daemons running on the node, which then introduce their own infrastructural complexity. Especially because of the same reasons why  #224 was rejected ( to keep us focused and keep things simple), I don't want to bloat our architecture with too many components.